tests/connectors/kubernetes_test.py

```diff
@@ -1130,3 +1134,140 @@ async def test_adjust_never_ready(self, config, kube: kubetest.client.TestClient
         with pytest.raises(AdjustmentRejectedError):
             description = await connector.adjust([adjustment])
             debug(description)
+
+##
+# Tests against an ArgoCD rollout
+# @pytest.mark.clusterrolebinding('cluster-admin') # May be needed
```

Comment:
You are probably getting away without having this enabled because it's a cluster-wide resource and probably hanging out from another test -- this is a weakness of the current setup where we are reliant on a single reused cluster that exposes us to confirmation bias.

Still marinating on when and how to do it but maybe we switch to dynamically allocated clusters so there's better isolation.

There's also a nascent concept of `system` tests where I want to take a small number of end to end tests (e.g., do your full rollout adjust flow) but parametrize it across specific environments like Argo versions, GKE/EKS/AKS, different Kubernetes versions, etc. so we get a real compatibility matrix for the big-ticket functionality